### PR TITLE
ci: add mechanism for versioning and automated npm publishing

### LINKED ISSUE #69: [REPO]: Automate publishing of packages
The Analog packages are versioned independently and this would automate the versioning and release of each package based on most recent commits.

Not sure if this would also include generating an updated changelog for each package. Ideally this could be triggered through a protected GitHub Action also

### PR DESCRIPTION
**Pull request by mainawycliffe**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [x] vite-angular-plugin
- [x] astro-angular
- [x] create-analog
- [x] router
- [x] platform
- [x] content

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Closes #69

### CODE REVIEW

**Comment by mainawycliffe** [comment]
@brandonroberts I need to figure out a way to test the workflows now

**Comment by mainawycliffe** [comment]
The release workflow should look something like this:

```sh
npx nx run router:version 
git push --folllow-tags
```

This should publish to NPM and create a GitHub Release with Changelog

**Comment by mainawycliffe** [comment]
@brandonroberts do we want to sync versions across packages?

**Comment by brandonroberts** [comment]
@mainawycliffe not at the moment. Maybe in the future. Also, some packages are built to go in the node_modules/@analogjs/{packageName} directory instead of dist/packages/{packageName}

**Comment by mainawycliffe** [comment]
> @mainawycliffe not at the moment. Maybe in the future. Also, some packages are built to go in the node_modules/@analogjs/{packageName} directory instead of dist/packages/{packageName}

@brandonroberts which packages are this? I can remove their versioning and also their tags as triggers in Github actions

**Comment by brandonroberts** [comment]
Every package except for create-analog

**Comment by brandonroberts** [comment]
To keep things straightforward, I would recommend just using the node_modules/@analogjs directories for publishing by default, and leaving the `create-analog` package as manual for now

**Comment by mainawycliffe** [comment]
So, in this case, we will have a single tag in the platform package - `@analog`  and this is going to build all other packages and then run npm publish from the  `node_modules/https://github.com/analogjs`  directory?

**Comment by brandonroberts** [comment]
No, still separate versions, just using `node_modules/@analogjs/{packageName}` for each package to publish, and exclude `create-analog` from automated publishing unless its a separate workflow.

**Comment by brandonroberts** [comment]
Thanks @mainawycliffe

File: packages/astro-angular/project.json [hunk 1]
```diff
@@ -30,11 +32,20 @@
...
       "options": {
-        "lintFilePatterns": ["packages/astro-angular/**/*.ts"]
+        "lintFilePatterns": [
+          "packages/astro-angular/**/*.ts"
+        ]
       }
     },
     "test": {
-      "executor": "@nx/vite:test"
+      "executor": "@nrwl/vite:test"
```
Comment (brandonroberts) [inline packages/astro-angular/project.json:41]:
```suggestion
      "executor": "@nx/vite:test"
```

File: packages/content/project.json [hunk 2]
```diff
@@ -34,7 +36,14 @@
     },
     "test": {
       "dependsOn": ["^build"],
-      "executor": "@nx/vite:test"
+      "executor": "@nrwl/vite:test"
```
Comment (brandonroberts) [inline packages/content/project.json:39]:
```suggestion
      "executor": "@nx/vite:test"
```

File: packages/create-analog/project.json [hunk 3]
```diff
@@ -6,14 +6,23 @@
...
+        "{options.outputPath}"
+      ],
       "options": {
         "outputPath": "dist/packages/create-analog",
         "command": "cpy packages/create-analog/* !packages/create-analog/node_modules/ !packages/create-analog/project.json !packages/create-analog/__tests__/ !packages/create-analog/yarn.lock dist/"
       }
     },
     "test": {
-      "executor": "@nx/vite:test"
+      "executor": "@nrwl/vite:test"
```
Comment (brandonroberts) [inline packages/create-analog/project.json:18]:
```suggestion
      "executor": "@nx/vite:test"
```

File: packages/platform/project.json [hunk 4]
```diff
@@ -39,12 +39,21 @@
...
-        "lintFilePatterns": ["packages/platform/**/*.ts"]
+        "lintFilePatterns": [
+          "packages/platform/**/*.ts"
+        ]
       }
     },
     "test": {
       "dependsOn": ["^build"],
-      "executor": "@nx/vite:test"
+      "executor": "@nrwl/vite:test"
```
Comment (brandonroberts) [inline packages/platform/project.json:49]:
```suggestion
      "executor": "@nx/vite:test"
```

File: packages/router/project.json [hunk 5]
```diff
@@ -31,7 +31,14 @@
       }
     },
     "test": {
-      "executor": "@nx/vite:test"
+      "executor": "@nrwl/vite:test"
```
Comment (brandonroberts) [inline packages/router/project.json:34]:
```suggestion
      "executor": "@nx/vite:test"
```

File: packages/vite-plugin-angular/project.json [hunk 6]
```diff
@@ -11,12 +11,21 @@
...
-        "assets": ["packages/vite-plugin-angular/*.md"],
+        "assets": [
+          "packages/vite-plugin-angular/*.md"
+        ],
         "updateBuildableProjectDepsInPackageJson": false
       }
     },
     "test": {
-      "executor": "@nx/vite:test"
+      "executor": "@nrwl/vite:test"
```
Comment (brandonroberts) [inline packages/vite-plugin-angular/project.json:21]:
```suggestion
      "executor": "@nx/vite:test"
```

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks @mainawycliffe! This will get us started on the way

**Pull request merged**